src/components/DocumentOptionsDropdown.tsx

```diff
@@ -24,24 +24,41 @@ import React from 'react'
 import styled from 'styled-components'
 
 import useExecCmd from '../hooks/use-exec-cmd'
+import { setManuscriptPrimaryLanguageCode } from '../lib/doc'
+import { useStore } from '../store/useStore'
+import DocumentLanguageSelector from './DocumentLanguageSelector'
 
-const VersionHistoryDropdown: React.FC = () => {
+const DocumentOptionsDropdown: React.FC = () => {
   const { isOpen, toggleOpen, wrapperRef } = useDropdown()
   const execCmd = useExecCmd()
 
+  const [storeState, , getState] = useStore((s) => ({
+    doc: s.doc,
+  }))
+
+  // Get selected language from document's primaryLanguageCode or default to 'en'
+  const selectedLanguage = storeState.doc?.attrs?.primaryLanguageCode || 'en'
+
+  const handleLanguageChange = async (languageCode: string) => {
+    const currentState = getState()
```

Comment:
getState is mostly used for the components for which is really hard to achieve proper reactivity. A more standard approach would be to use 'view' from the store and rerender/recreate functions and components when the value in the store changes.